src/org/stringtemplate/v4/Interpreter.java

```diff
@@ -414,7 +414,8 @@ protected int _exec(STWriter out, InstanceScope scope) {
 					break;
 				case Bytecode.INSTR_NEWLINE :
 					try {
-						if ( prevOpcode==Bytecode.INSTR_NEWLINE ||
+						if ( (prevOpcode==0 && !self.isAnonSubtemplate() && !self.impl.isRegion) ||
```

Comment:
:thought_balloon: I find it strange that `Misc.trimOneStartingNewline` and `Misc.trimOneTrailingNewline` are used for template definitions, but not for regions or anonymous subtemplates. It seems like we could simplify the new condition to just `prevOpcode==0` if we applied the trim operation to subtemplates.